docs/site/HasMany-relation.md

```diff
@@ -178,6 +180,42 @@ export interface OrderRelations {
 export type OrderWithRelations = Order & OrderRelations;
 ```
 
+### Relation Metadata
+
+`keyFrom`, `keyTo` and `name` fields in the `hasMany` relation matadata allow us
+to customize and specify our id/foreign key names:
+
+- `keyFrom`: the primary key of the source model. The default value is the id
+  property in the source model, which is `id` property of `Customer` in our
+  example.
+- `keyTo`: the foreign key of the target model. The default value is the source
+  model name appended with `id` in camel case, which is the `customerId`
+  property of `Order` in our example.
+- `name`: the name of the relation, which is the `orders` property of `Customer`
+  in our example.
+
+We can specify the `keyTo` field via `@hasMany` decorator if we have customized
+the foreign key name as `my_customer_id` instead of `customerId`:
+
+```ts
+@model()
+export class Order extends Entity {
+  // constructor, properties, etc.
+  @belongsTo(() => Customer)
+  my_customer_id: number; // customized foreign key name
```

Comment:
In this example, I believe the relation definition will use the name `my_customer_id`. This will create problems when including related models, because we will replace `my_customer_id` property from the numeric id to the actual customer instance.

I am proposing to fix the relation name.

```ts
@belongsTo(() => Customer, {name: 'customer'})
my_customer_id: number; // customized foreign key name
```

Reply:
Right mb I forgot to add `name`. 

I am thinking that to have two cases in this doc:
- models only have one relation: hasMany or hasOne, no belongsTo
  - the foreign key is not decorated by `@belongsTo`, we don't need to modify property metadata in another model (`Order` in this case)
  - simple usage to show the idea of `keyFrom, keyTo`
- models have hasMany/hasOne and belongsTo:
  - modify metadata in both models to customize the names.
  - the case to show what the changes it needs to make if we have such relations

I don't know if it's necessary to have two cases, thoughts? @bajtos

Reply:
I'm wondering... if we have case #2 you described above, doesn't it mean we have case #1 covered too? Or is there anything we'd need to do differently?

Reply:
They are a bit different from my understanding. 
For case #1, the change is only needed within `hasMany` relation. 
For case #2, we are modifying `belongsTo` relation as well ([here](https://github.com/strongloop/loopback-next/pull/3805/files#diff-65c8c6f0fd2bb35d13770c5822272740R227)). 

When I just started using LB4, it was not clear to me that if I can use hasMany without belongsTo cause we use them together in every example. So I want to make it clear in the docs. I am ok to just keep #2 if it's not necessary.

Reply:
My opinion: If we want to show how to use HasMany without BelongsTo, then we should make that intent explicit. Create a new section, explain that it's possible to use one relation without the other and offer an example how to do it.

Then in the rest of the documentation, I'd prefer to stay consistent with the prevalent approach, which is to describe both relations in tandem.

Reply:
yes! in `hasMany`, we introduced the usages of `hasMany` without and with `belongsTo`. So I think it should be fine to have two `keyTo` cases mapping with those two usages.